tests/unit/test_parts.py

```diff
@@ -335,6 +335,31 @@ def test_part_has_overlay(self, partitions):
         p = Part("foo", {}, partitions=partitions)
         assert p.has_overlay is False
 
+    @pytest.mark.parametrize(
+        ("tc_spec", "tc_result"),
+        [
+            ({}, False),
+            ({"stage-packages": ["base-files_base", "hello_bins"]}, True),
+            ({"stage-packages": ["bash", "python3"]}, False),
+        ],
+    )
+    def test_part_has_slices(self, partitions, tc_spec, tc_result):
+        p = Part("foo", tc_spec, partitions=partitions)
+        assert p.spec.has_slices == tc_result
+
+    @pytest.mark.parametrize(
+        ("tc_spec", "tc_result"),
+        [
+            ({}, False),
+            ({"build-snaps": ["hello", "chisel/latest/candidate"]}, True),
+            ({"build-snaps": ["chisel"]}, True),
+            ({"build-snaps": ["hello"]}, False),
```

Comment:
You had a good point in your comment, I think you should test it.

```suggestion
            ({"build-snaps": ["hello"]}, False),
            ({"build-snaps": ["chiselhelper"]}, False),
```

Reply:
Added.